sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastNestedLoopJoinExecBase.scala

```diff
@@ -558,6 +558,15 @@ abstract class GpuBroadcastNestedLoopJoinExecBase(
     broadcastExchange.executeColumnarBroadcast[Any]()
   }
 
+  private def isUnconditionalJoin(condition: Option[GpuExpression]): Boolean = {
+    condition.forall {
+      case GpuLiteral(true, BooleanType) =>
```

Comment:
nit: do we want to use https://github.com/NVIDIA/spark-rapids/blob/4a01c1a18ec67ff85377451ee0695a3b1f232902/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala#L571 to allow it to also have an alias in it. Not that I think it would ever happen, but who knows.

Reply:
We can't use GpuOverrides.extractLit here since it only works on CPU expressions (i.e.: Literal and Alias instead of GpuLiteral and GpuAlias), but I agree it's probably worth checking for aliases just in case.